packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2140,6 +2140,12 @@ class _WidgetInspectorState extends State<WidgetInspector>
   void initState() {
     super.initState();
 
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      // Set the root render object so that the inspector
+      // layer can apply the correct transforms
+      selection.rootRenderObject = context.findRenderObject();
```

Comment:
adding this with a post frame callback concerns me because there may now be a 1 frame lag where the render object is inconsistent for the selection and the inspector overlay. Can we update the root render object earlier to avoid this issue?

Reply:
For example. maybe instead have _RenderInspectorOverlay check what its immediate parent is.

Reply:
I ended up passing the _RenderInspectorOverlay parent to the layer.